include/beman/execution26/detail/split.hpp

```diff
@@ -0,0 +1,375 @@
+// include/beman/execution26/detail/split.hpp                         -*-C++-*-
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef INCLUDED_BEMAN_EXECUTION26_DETAIL_SPLIT
+#define INCLUDED_BEMAN_EXECUTION26_DETAIL_SPLIT
+
+#include <beman/execution26/detail/atomic_intrusive_stack.hpp>
+#include <beman/execution26/detail/connect_result_t.hpp>
+#include <beman/execution26/detail/default_impls.hpp>
+#include <beman/execution26/detail/emplace_from.hpp>
+#include <beman/execution26/detail/error_types_of_t.hpp>
+#include <beman/execution26/detail/completion_signatures_for.hpp>
+#include <beman/execution26/detail/get_domain_early.hpp>
+#include <beman/execution26/detail/get_stop_token.hpp>
+#include <beman/execution26/detail/child_type.hpp>
+#include <beman/execution26/detail/completion_signatures_of_t.hpp>
+#include <beman/execution26/detail/sender_for.hpp>
+#include <beman/execution26/detail/impls_for.hpp>
+#include <beman/execution26/detail/inplace_stop_source.hpp>
+#include <beman/execution26/detail/make_sender.hpp>
+#include <beman/execution26/detail/type_list.hpp>
+#include <beman/execution26/detail/meta_combine.hpp>
+#include <beman/execution26/detail/meta_unique.hpp>
+#include <beman/execution26/detail/receiver.hpp>
+#include <beman/execution26/detail/stop_callback_for_t.hpp>
+#include <beman/execution26/detail/stop_token_of_t.hpp>
+#include <beman/execution26/detail/stoppable_token.hpp>
+#include <beman/execution26/detail/value_types_of_t.hpp>
+
+#include <atomic>
+#include <optional>
+#include <variant>
+#include <tuple>
+
+// ----------------------------------------------------------------------------
+
+namespace beman::execution26::detail {
+
+struct split_impl_t {};
+template <>
+struct impls_for<split_impl_t> : ::beman::execution26::detail::default_impls {
+
+    template <class Sndr>
+    struct shared_state;
+
+    struct split_env {
+        ::beman::execution26::inplace_stop_source* stop_source;
+
+        ::beman::execution26::inplace_stop_token query(::beman::execution26::get_stop_token_t) const noexcept {
+            return stop_source->get_token();
+        }
+    };
+
+    struct local_state_base {
+        local_state_base* next{nullptr};
+        virtual auto      notify() noexcept -> void = 0;
+
+        local_state_base() noexcept                                  = default;
+        local_state_base(const local_state_base&)                    = delete;
+        local_state_base(local_state_base&&)                         = delete;
+        auto operator=(const local_state_base&) -> local_state_base& = delete;
+        auto operator=(local_state_base&&) -> local_state_base&      = delete;
```

Comment:
I have recently introduced a base class [`::beman::execution26::detail::virtual_immovable`](https://github.com/bemanproject/execution26/blob/7beac1618de0cf7a5993742b71bd51cc422668ae/include/beman/execution26/detail/immovable.hpp#L23) which provides this boiler plate: some of the `clang-tidy` checks insisted in that stuff being written which is quite annoying. The class is currently possibly in the wrong header (`immovable.hpp`).

Reply:
Note that I think it is more "efficient" to define a protected destructor and make the derived class final. But I can just switch to virtual immovable. 

NB. I run clang-tidy and cppcheck locally and it still complains on classes using virtual_immovable. Although I have to check which one complains. Would be good to use the same tooling consistently in CI and local dev.